src/pkg/backup/details/details_test.go

```diff
@@ -303,7 +331,83 @@ func (suite *DetailsUnitSuite) TestDetails_AddFolders() {
 
 			for _, e := range deets.Entries {
 				assert.Contains(t, test.expectedShortRefs, e.ShortRef)
+				assert.Equal(t, test.expectedFolderInfo[e.ShortRef].Size, e.Folder.Size)
+				assert.Equal(t, test.expectedFolderInfo[e.ShortRef].Modified, e.Folder.Modified)
 			}
 		})
 	}
 }
+
+func (suite *DetailsUnitSuite) TestDetails_AddFoldersDifferentServices() {
+	itemTime := time.Date(2022, 10, 21, 10, 0, 0, 0, time.UTC)
+	folderTimeOlderThanItem := time.Date(2022, 9, 21, 10, 0, 0, 0, time.UTC)
+
+	table := []struct {
+		name               string
+		item               details.ItemInfo
+		expectedFolderInfo details.FolderInfo
+	}{
+		{
+			name: "Exchange",
+			item: details.ItemInfo{
+				Exchange: &details.ExchangeInfo{
+					Size:     20,
+					Modified: itemTime,
+				},
+			},
+			expectedFolderInfo: details.FolderInfo{
+				Size:     20,
+				Modified: itemTime,
+			},
+		},
+		{
+			name: "OneDrive",
+			item: details.ItemInfo{
+				OneDrive: &details.OneDriveInfo{
+					Size:     20,
+					Modified: itemTime,
+				},
+			},
+			expectedFolderInfo: details.FolderInfo{
+				Size:     20,
+				Modified: itemTime,
+			},
+		},
+		{
+			name: "SharePoint",
+			item: details.ItemInfo{
+				SharePoint: &details.SharePointInfo{
+					Size:     20,
+					Modified: itemTime,
+				},
+			},
+			expectedFolderInfo: details.FolderInfo{
+				Size:     20,
+				Modified: itemTime,
+			},
+		},
+	}
+	for _, test := range table {
+		suite.T().Run(test.name, func(t *testing.T) {
+			folderEntry := details.FolderEntry{
+				RepoRef:   "rr1",
+				ShortRef:  "sr1",
+				ParentRef: "pr1",
+				Info: details.ItemInfo{
+					Folder: &details.FolderInfo{
+						Modified: folderTimeOlderThanItem,
```

Comment:
As a sanity check, maybe not set Modified at all (because the first time a folder is added, it won't have it set). Should work, given the default value of time.Time, but for added confidence.